### PR TITLE
i18n: Add Thai language

### PR DESCRIPTION
**Pull request by weeix**
This PR is intended to incorporate the Thai language into Keycloak, specifically in the following Themes:

- Base > Account
- Base > Email
- Base > Login
- Keycloak.v2

### CODE REVIEW

**Comment by jonkoops** [comment]
Hi @weeix, thanks for your PR. We require changes to localization to have at least one reviewer which is a native speaker of the language. If you know someone who can review your work here, please ask them to do so.

**Comment by weeix** [comment]
Hello @jonkoops, I have shared invitations for reviewing this PR in several Facebook groups. These groups consist of Thai individuals who may potentially utilize Keycloak in their professional capacities.

**Review by nickethachin — APPROVED** [review]
I'm Thai native speaker and theses translation look good to me.
Both context and meaning is correct and to the point.

**Review by ghost — APPROVED** [review]
I’m Thai and this commit looks good to me.

**Review by ahus1 — DISMISSED** [review]
Approving as of community reviews

**Comment by amincheloh** [comment]
According to the Royal Society of Thailand (ราชบัณฑิตยสภา), the word 'Application' should be transliterated as 'แอปพลิเคชัน'.

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+role_manage-identity-providers=จัดการผู้ให้บริการตัวตน
+role_manage-clients=จัดการไคลเอนต์
+role_manage-events=จัดการเหตุการณ์
+role_view-profile=ดูโปรไฟล์
+role_manage-account=จัดการบัญชี
+role_manage-account-links=จัดการลิงก์บัญชี
+role_manage-consent=จัดการการยินยอม
+role_read-token=อ่านโทเค็น
+role_offline-access=การเข้าถึงแบบออฟไลน์
+role_uma_authorization=ขออนุญาติ
```
Comment (amincheloh) [inline themes/src/main/resources-community/theme/base/account/messages/messages_th.properties:101]: ขออนุญาต

**Comment by ahus1** [comment]
@weeix - I see additional comments from the community. Please let me know if you want to address them in this pull request or a future pull request.

**Comment by weeix** [comment]
> @weeix - I see additional comments from the community. Please let me know if you want to address them in this pull request or a future pull request.

I will address them in this pull request.

**Comment by sapondanaisriwan** [comment]
แอพพลิเคชัน -> แอปพลิเคชัน

File: themes/src/main/resources-community/theme/keycloak.v2/account/messages/messages_th.properties [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+backToAdminConsole=กลับไปที่คอนโซลผู้ดูแลระบบ
+accountManagementWelcomeMessage=ยินดีต้อนรับสู่การจัดการบัญชี Keycloak
+
+# Personal info page
+personalInfoHtmlTitle=ข้อมูลส่วนตัว
+
+# Device activity page
+signedInDevices=อุปกรณ์ที่ลงชื่อเข้าใช้
+signedInDevicesExplanation=ลงชื่อออกจากอุปกรณ์ที่ไม่คุ้นเคยใด ๆ
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/keycloak.v2/account/messages/messages_th.properties:33]: should be nice with `ลงชื่ออกจากอุปกรณ์อื่นที่ไม่คุ้นเคย`

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+role_manage-account=จัดการบัญชี
+role_manage-account-links=จัดการลิงก์บัญชี
+role_manage-consent=จัดการการยินยอม
+role_read-token=อ่านโทเค็น
+role_offline-access=การเข้าถึงแบบออฟไลน์
+role_uma_authorization=ขออนุญาต
+client_account=บัญชี
+client_account-console=คอนโซลบัญชี
+client_security-admin-console=คอนโซลผู้ดูแลระบบความปลอดภัย
+client_admin-cli=Admin CLI
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/base/account/messages/messages_th.properties:105]: CLI สำหรับผู้ดูแลระบบ

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+client_realm-management=การจัดการ realm
+client_broker=Broker
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/base/account/messages/messages_th.properties:107]: may be `ตัวแทน` ?

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties [hunk 5]
```diff
@@ -0,0 +1,383 @@
...
+totpManualStep3=ใช้ค่าการกำหนดค่าต่อไปนี้หากแอปพลิเคชันอนุญาตให้ตั้งค่า:
+totpUnableToScan=ไม่สามารถสแกนได้?
+totpScanBarcode=สแกนบาร์โค้ด?
+
+totp.totp=Time-based
+totp.hotp=Counter-based
+
+totpType=ประเภท
+totpAlgorithm=ขั้นตอนวิธี
+totpDigits=Digits
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/base/account/messages/messages_th.properties:162]: ตัวเลข

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties [hunk 6]
```diff
@@ -0,0 +1,383 @@
...
+totpInterval=Interval
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/base/account/messages/messages_th.properties:163]: ช่วงเวลา

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties [hunk 7]
```diff
@@ -0,0 +1,383 @@
...
+doRemoveRequest=ลบคำขอ
+peopleAccessResource=ผู้ที่มีสิทธิ์เข้าถึงทรัพยากรนี้
+resourceManagedPolicies=การอนุญาตให้เข้าถึงทรัพยากรนี้
+resourceNoPermissionsGrantingAccess=ไม่มีการอนุญาตให้เข้าถึงทรัพยากรนี้
+anyAction=การกระทำใด ๆ
+description=คำอธิบาย
+name=ชื่อ
+scopes=ขอบเขต
+resource=ทรัพยากร
+user=ผู้ใช้
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/base/account/messages/messages_th.properties:252]: `user` should be nice with `ผู้ใช้งาน`

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties [hunk 8]
```diff
@@ -0,0 +1,383 @@
...
+peopleSharingThisResource=ผู้ที่แบ่งปันทรัพยากรนี้
+shareWithOthers=แบ่งปันกับผู้อื่น
+needMyApproval=ต้องได้รับการอนุมัติจากฉัน
+requestsWaitingApproval=คำขอของคุณที่รอการอนุมัติ
+icon=ไอคอน
+requestor=ผู้ขอ
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/base/account/messages/messages_th.properties:258]: ผู้ร้องขอ

File: themes/src/main/resources-community/theme/base/login/messages/messages_th.properties [hunk 9]
```diff
@@ -0,0 +1,493 @@
...
+loginTotpManualStep2=เปิดแอปพลิเคชันและใส่คีย์:
+loginTotpManualStep3=ใช้การตั้งค่าต่อไปนี้หากแอปพลิเคชันสามารถระบุการตั้งค่าได้:
+loginTotpUnableToScan=ไม่สามารถสแกนได้?
+loginTotpScanBarcode=สแกนบาร์โค้ด?
+loginCredential=Credential ข้อมูลประจำตัว
+loginOtpOneTime=รหัสสำหรับใช้ครั้งเดียว
+loginTotpType=ประเภท
+loginTotpAlgorithm=ขั้นตอนวิธี
+loginTotpDigits=Digits
+loginTotpInterval=Interval
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/base/login/messages/messages_th.properties:134]: ช่วงเวลา

**Review by bouroo — CHANGES_REQUESTED** [review]

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties (hunk 5 above)
Comment (weeix) [reply themes/src/main/resources-community/theme/base/account/messages/messages_th.properties:162]: หรือเป็น หลัก ดีไหมครับ ไปเจอภาพในเน็ตว่ามันมีให้เลือกระหว่าง 6 หรือ 8 หลัก

https://www.keycloak.org/docs/latest/server_admin/images/otp-policy.png

**Comment by weeix** [comment]
Thank you. Additional suggestions are always welcome.

Changes (force-pushed):

- Translations have been enhanced based on community review suggestions.
- All changes have been consolidated into a single commit.
- The `totpAppMicrosoftAuthenticatorName` translation, previously missing, has now been added to the *Base > Account* section.
- The `access-denied-when-idp-auth` translation, previously missing, has now been added to the *Base > Account* section.
- The `totpAppMicrosoftAuthenticatorName` translation, previously missing, has now been added to the *Base > Login* section.
- A newline has been inserted in the *Base > Account* section for alignment with the English translation.
- All trailing spaces have been removed.

**Review by chalut01 — APPROVED** [review]
LGTM

**Comment by amincheloh** [comment]
LGTM

File: themes/src/main/resources-community/theme/base/account/messages/messages_th.properties (hunk 5 above)
Comment (bouroo) [reply themes/src/main/resources-community/theme/base/account/messages/messages_th.properties]: ก็ดีนะครับ พอดีผมไม่เห็นว่าบริบทจริง ๆ เป็นยังไง

**Review by ahus1 — APPROVED** [review]
Thank you for these changes and the review from the community!

File: themes/src/main/resources-community/theme/base/login/messages/messages_th.properties [hunk 10]
```diff
@@ -0,0 +1,493 @@
...
+proceedWithAction=&raquo; คลิกที่นี่เพื่อดำเนินการ
+acrNotFulfilled=คุณสมบัติไม่เป็นไปตามข้อกำหนดในการพิสูจน์ตัวจริง
+
+requiredAction.CONFIGURE_TOTP=ตั้งค่า OTP
+requiredAction.terms_and_conditions=ข้อกำหนดและเงื่อนไข
+requiredAction.UPDATE_PASSWORD=ปรับปรุงรหัสผ่าน
+requiredAction.UPDATE_PROFILE=ปรับปรุงโปรไฟล์
+requiredAction.VERIFY_EMAIL=ยืนยันอีเมล
+requiredAction.CONFIGURE_RECOVERY_AUTHN_CODES=สร้างรหัสกู้คืน
+requiredAction.webauthn-register-passwordless=Webauthn Register Passwordless
```
Comment (bouroo) [inline themes/src/main/resources-community/theme/base/login/messages/messages_th.properties:377]: ลงทะเบียนแบบไม่ใช้รหัสผ่านด้วย WebAuthn

**Comment by jonkoops** [comment]
Giving this a quick rebase to hopefully pass the CI.

**Review by jonkoops — APPROVED** [review]

**Comment by ahus1** [comment]
Thank you for this PR. As there were some additional comments which weren't included in this PR, please create additional PRs as needed to address those comments. We'll be happy to merge them once they - like this one - have been reviewed by native speaker.

**Pull request merged**